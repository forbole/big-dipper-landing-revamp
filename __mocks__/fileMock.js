// __mocks__/fileMock.js
module.exports = {
  blurDataURL: 'data:image/png;base64,imagedata',
  height: 24,
  src: '/img.jpg',
  width: 24,
}