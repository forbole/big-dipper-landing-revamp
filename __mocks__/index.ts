export { default as MockTheme } from "./theme";
