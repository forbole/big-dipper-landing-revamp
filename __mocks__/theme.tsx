/* eslint-disable */
import { ThemeProvider } from '@mui/material/styles';
import { ReactNode } from 'react';
import { darkTheme } from '@/src/styles/theme';

/**
 * Theme mocker to handle custom keys
 */
const MockTheme = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={darkTheme}>{children}</ThemeProvider>;
};

export default MockTheme;
