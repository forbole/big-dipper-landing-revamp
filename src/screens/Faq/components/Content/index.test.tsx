/* eslint-disable react/display-name */
import { MockTheme } from "@/__mocks__";
import Box from "@mui/material/Box";
import { render, screen } from "@testing-library/react";

import Component from ".";

jest.mock("@/src/components/ContentBox", () => (props: object) => (
  <Box data-testid="ContentBox" {...props} />
));

// ==================================
// unit tests
// ==================================
describe("Content", () => {
  it("matches snapshot", async () => {
    const { container } = render(
      <MockTheme>
        <Component
          content={[
            {
              answer: "Fine.",
              question: "How are you?",
            },
          ]}
        />
      </MockTheme>,
    );

    expect(screen.getByText(/Fine./i)).toBeInTheDocument();
    expect(screen.getByTestId("ContentBox")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
