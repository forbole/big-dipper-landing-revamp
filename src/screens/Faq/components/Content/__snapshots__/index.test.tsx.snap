// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content matches snapshot 1`] = `
.emotion-0 .faqcontent__list {
  margin-top: 16px;
}

.emotion-0 .faqcontent__list-item:not(:last-child) {
  margin-bottom: 24px;
}

.emotion-0 .faqcontent__list-title {
  color: #F6504B;
  margin-bottom: 8px;
  margin-top: 40px;
}

@media (max-width:767.95px) {
  .emotion-0 .faqcontent__list-title {
    font-size: 1rem;
  }
}

.emotion-0 .faqcontent__list-title-bullet {
  color: #0939B0;
  margin-right: 8px;
}

.emotion-0 .faqcontent__question {
  color: #F6504B;
  margin-bottom: 16px;
}

@media (max-width:767.95px) {
  .emotion-0 .faqcontent__question {
    font-size: 1rem;
  }
}

.emotion-0 .faqcontent__wrapper:not(:last-child) {
  margin-bottom: 96px;
}

@media (max-width:767.95px) {
  .emotion-0 .faqcontent__wrapper:not(:last-child) {
    margin-bottom: 72px;
  }
}

@media (max-width:767.95px) {
  .emotion-0 .MuiTypography-body1 {
    font-size: 0.875rem;
  }
}

.emotion-1 {
  margin: 0;
  font-size: 1.125rem;
  font-weight: 600;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  line-height: 1.334;
}

.emotion-2 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  font-family: 'SF Pro Text',-apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    data-testid="ContentBox"
  >
    <div
      class="faqcontent__wrapper"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 faqcontent__question emotion-1"
      >
        How are you?
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-2"
      >
        Fine.
      </p>
    </div>
  </div>
</div>
`;
